### PR TITLE
refactor: depreate api.Context for playbook and upstream

### PR DESCRIPTION
**Pull request by yashmehrotra**

### CODE REVIEW

File: playbook/controllers.go [hunk 1]
```diff
@@ -200,7 +201,7 @@ func HandlePlaybookList(c echo.Context) error {
 }
 
 func HandlePlaybookRunApproval(c echo.Context) error {
-	ctx := c.(api.Context)
+	ctx := c.Request().Context().(context.Context)
```
Comment (moshloop) [inline playbook/controllers.go:204]: Weren't we going to wrap the echo context ?

Comment (yashmehrotra) [reply playbook/controllers.go:204]: We are wrapping it in the middleware: https://github.com/flanksource/incident-commander/blob/main/cmd/server.go#L77

Just need to type-assert in the funcs, or should I replace this with wrap as well ?

Comment (yashmehrotra) [reply playbook/controllers.go:204]:
```go
ctx := api.DefaultContext.Wrap(c.Request().Context())

// or

ctx := c.Request().Context().(context.Context)
```

Comment (yashmehrotra) [reply playbook/controllers.go:204]: actually, we have to do it in the middleware itself cause the auth middleware needs duty.Context

**Comment by adityathebe** [comment]
@yashmehrotra I think you've covered almost all controllers _(apart from maybe just the agent's controller)_. Why not get rid of `api.Context` totally from the middleware in this PR itself?

`func createHTTPServer` would accept duty's context in that case.

File: playbook/controllers.go (hunk 1 above)
Comment (adityathebe) [reply playbook/controllers.go:204]: > We are wrapping it in the middleware: https://github.com/flanksource/incident-commander/blob/main/cmd/server.go#L77
> 
> Just need to type-assert in the funcs, or should I replace this with wrap as well ?

The way we are wrapping contexts in middleware is very confusing

`c.SetRequest(c.Request().WithContext(api.DefaultContext.Wrap(c.Request().Context())))`

**Comment by yashmehrotra** [comment]
> @yashmehrotra I think you've covered almost all controllers (apart from maybe just the agent's controller). Why not get rid of api.Context totally from the middleware in this PR itself?

@adityathebe Doing that in a different PR with other changes, wanted to keep the scope smaller for this one

File: playbook/controllers.go (hunk 1 above)
Comment (yashmehrotra) [reply playbook/controllers.go:204]: Should we create a func around it ?
```go
func WithEchoContext(c echo.Context, ctx dutycontext.Context) *http.Request {

    return c.Request().WithContext(
      ctx.Wrap(c.Request().Context())
  )
  ```
  
  And do  this in middleware ?
  ```go

c.SetRequest(WithEchoContext(c, api.DefaultContext))
```

Comment (adityathebe) [reply playbook/controllers.go:204]: @yashmehrotra Not sure. We're still doing the same thing but in a different place. But also there doesn't seem to be a cleaner way to inject our context into echo's context.

We also could use `c.Set("duty-context", api.DefaultContext.Wrap(c.Request().Context())` although maybe that's not any better.

Comment (yashmehrotra) [reply playbook/controllers.go:204]: > But also there doesn't seem to be a cleaner way to inject our context into echo's context.
yeah, i think lets leave it to original approach as its explicit

**Pull request merged**